packages/synchro-charts/src/components/sc-dial/sc-dial.tsx

```diff
@@ -0,0 +1,161 @@
+
+const renderCell: RenderCell = props => <sc-dial-base {...props} />;
+
+const DEFAULT_MARGINS: Partial<SizeConfig> = {
+  marginLeft: 10,
+  marginTop: 0,
+  marginBottom: DEFAULT_CHART_CONFIG.size.marginBottom,
+  marginRight: 5,
+};
+
+// Fits two rows of legend rows
+const THRESHOLD_LEGEND_HEIGHT_PX = 50;
+@Component({
+  tag: 'sc-dial',
+  styleUrl: 'sc-dial.css',
+  shadow: false,
+})
+export class ScDial implements DialConfig {
+  @Prop() viewport: ViewPortConfig;
+  @Prop() widgetId!: string;
+  @Prop() dataStream!: DataStream;
+  @Prop() associatedStreams?: StreamAssociation[];
+  @Prop() annotations?: Annotations;
+  @Prop() messageOverrides: MessageOverrides = {};
+
+  @Prop() size?: MinimalSizeConfig = { width: 276, height: 276 };
+
+  @Event()
+  widgetUpdated: EventEmitter<WidgetConfigurationUpdate>;
+
+  /** Widget data stream names */
+  @State() names: NameValue[] = [];
+
+  componentWillRender() {
+    validate(this);
+  }
+
+  getPoint = (dataStream: DataStream): DataPoint | undefined => {
+    if (dataStream.data && dataStream.data.length > 0) {
+      const point = dataStream.data[dataStream.data.length - 1];
+      return point;
+    }
+
+    return undefined;
+  };
+
+  getAlarmStream = (dataStream: DataStream): boolean => {
+    return !!this.associatedStreams && this.associatedStreams.some(stream => stream.id === dataStream.id);
+  };
+
+  getBreachedThreshold = (point: DataPoint | undefined, dataStream: DataStream): Threshold | undefined =>
+    breachedThreshold({
+      value: point && point.y,
+      date: isMinimalStaticViewport(this.viewport) ? new Date(this.viewport.end) : new Date(),
+      dataStreams: [this.dataStream],
+      dataStream,
+      thresholds: getThresholds(this.annotations),
+    });
+
+  /**
+   * Emit the current widget configuration
+   */
+  emitUpdatedWidgetConfiguration = (dataStreams?: DataStream[]): void => {
+    const configUpdate: WidgetConfigurationUpdate = {
+      movement: undefined,
+      scale: undefined,
+      layout: undefined,
+      legend: undefined,
+      annotations: this.annotations,
+      axis: undefined,
+      widgetId: this.widgetId,
+      dataStreams: dataStreams ? getDataStreamForEventing(dataStreams) : [this.dataStream],
+    };
+    this.widgetUpdated.emit(configUpdate);
+  };
+
+  /**
+   * On Widget Updated - Persist `DataStreamInfo`
+   *
+   * Emits an event which persists the current `NameValue[]` state into the
+   * data stream info.
+   */
+  onWidgetUpdated() {
+    // Construct the config update with the new names specified.
+    const nameValue = this.names.find(({ id: nameId }) => this.dataStream.id === nameId);
+    const name = nameValue != null ? nameValue.name : this.dataStream.name;
+    const updatedDataStream = { ...this.dataStream, name };
+    this.emitUpdatedWidgetConfiguration([updatedDataStream]);
+  }
+
+  onChangeLabel = ({ streamId, name }: { streamId: string; name: string }): void => {
```

Comment:
i'm confused, is this called anywhere? if not, please remove it

Reply:
Remove onChangeLabel function